File: ExpenseManager.ClientApplication/Scripts/Data/RegisterModel.js
export default class RegisterModel {
  constructor(fullName='', companyName='', phoneNumber='', email='', password='') {
    this.fullName = fullName;
    this.companyName = companyName;
    this.phoneNumber = phoneNumber;
    this.email = email;
    this.password = password;
  }
}

File: ExpenseManager.Web/Authentication/Models/RegisterModel.cs
using System.ComponentModel.DataAnnotations;

namespace ExpenseManager.Models.Authentication.Models
{
    public class RegisterModel
    {
        [Required(ErrorMessage = "Full name is required")]
        public string FullName { get; set; }

        [Required(ErrorMessage = "Company name is required")]
        public string CompanyName { get; set; }

        [Required(ErrorMessage = "Phone number is required")]
        public string PhoneNumber { get; set; }
  
        [EmailAddress]  
        [Required(ErrorMessage = "Email is required")]
        public string Email { get; set; }
  
        [Required(ErrorMessage = "Password is required")]
        public string Password { get; set; }
    }
}